front/app/components/admin/PostManager/InputManager.tsx

```diff
@@ -0,0 +1,376 @@
+    // otherwise the if condition will always pass.
+    const topicIdsSet = topicIds.length > 0 ? new Set(topicIds) : undefined;
+
+    if (topicIdsSet) {
+      return ideaTopics?.data.filter((topic) => topicIdsSet?.has(topic.id));
+    }
+
+    return ideaTopics?.data;
+  };
+
+  const topicsData = getTopicsData();
+  const [search] = useSearchParams();
+  const [selection, setSelection] = useState<Set<string>>(new Set());
+
+  const [activeFilterMenu, setActiveFilterMenu] =
+    useState<TFilterMenu>(defaultFilterMenu);
+  const [previewPostId, setPreviewPostId] = useState<string | null>(null);
+  const [previewMode, setPreviewMode] = useState<PreviewMode>('view');
+
+  useEffect(() => {
+    if (search.get('selected_idea_id')) {
+      setPreviewMode('view');
+      setPreviewPostId(search.get('selected_idea_id'));
+    }
+    removeSearchParams(['selected_idea_id']);
+  }, [search]);
+
+  useEffect(() => {
+    setActiveFilterMenu((activeFilterMenu) => {
+      if (
+        !visibleFilterMenus.find((item) => item === activeFilterMenu) &&
+        visibleFilterMenus[0]
+      ) {
+        return visibleFilterMenus[0];
+      } else {
+        return activeFilterMenu;
+      }
+    });
+  }, [visibleFilterMenus]);
+
+  if (!ideas || !topicsData) return null;
+
+  // Filtering handlers
+  const getSelectedProject = () => {
+    return Array.isArray(queryParameters.projects) &&
+      queryParameters.projects.length === 1
+      ? queryParameters.projects[0]
+      : undefined;
+  };
+
+  const handleOnChangeProjects = (projectIds: string[] | undefined) => {
+    if (type !== 'AllIdeas') return;
+
+    const accessibleProjectsIds = projects
+      ? projects.map((project) => project.id)
+      : null;
+
+    if (!projectIds || projectIds.length === 0) {
+      accessibleProjectsIds && onChangeProjects(accessibleProjectsIds);
+    } else {
+      onChangeProjects(projectIds);
+    }
+  };
+  // End filtering hanlders
+
+  const resetSelection = () => {
+    setSelection(new Set());
+  };
+  // End selection management
+
+  // Filter menu
+  const handleChangeActiveFilterMenu = (activeFilterMenu: TFilterMenu) => {
+    setActiveFilterMenu(activeFilterMenu);
+  };
+  // End Filter menu
+
+  // Modal Preview
+  const openPreview = (postId: string) => {
+    setPreviewPostId(postId);
+    setPreviewMode('view');
+  };
+
+  const openPreviewEdit = () => {
+    const previewPostId = [...selection][0];
+    if (selection.size === 1 && previewPostId) {
+      setPreviewMode('edit');
+      setPreviewPostId(previewPostId);
+    }
+  };
+
+  const switchPreviewMode = () => {
+    if (previewMode === 'edit') {
+      setPreviewMode('view');
+    } else {
+      setPreviewMode('edit');
+    }
+  };
+
+  const closePreview = () => {
+    setPreviewPostId(null);
+    setPreviewMode('view');
+  };
+  // End Modal Preview
+
+  const onChangeTopics = (topics: string[]) => {
+    setQueryParameters({ ...queryParameters, 'page[number]': 1, topics });
+  };
```

Comment:
Here too wondering if these should be part of useIdeas (see other comment).

Reply:
I think it's better to keep the hook clean and handle these inside of the component.